### PR TITLE
[NNVM] Bug fix Prevent fusing convolution with injective op 

### PR DESCRIPTION
**Pull request by masahi**
Continuing from #1603.

Modify operator fuser to include following logics.

* During the initial partition step, if injective op is followed by broadcast op, mark broadcast op's op pattern to be injective
* In the last grouping step, check for input nodes op patterns. If one of them is kOutEWiseFusable and the other is kInjective, ignore the kOutEWiseFusable node and fuse only kInjective node.

A test case is included. This test fails without this PR.
I confirmed that this change does not affect compling resnet, mobilenet and vgg.

@tqchen please check if the logic is good.

### CODE REVIEW

**Comment by tqchen** [comment]
@merrymercy can you also review this?

File: nnvm/src/compiler/graph_fuse.cc [hunk 1]
```diff
@@ -135,8 +141,25 @@ nnvm::Graph GraphFindFusibleGroups(nnvm::Graph g) {
     if (group_vec[nid] == -1) {
       group_vec[nid] = nid;
     }
+
+    // Check if injective op and out_ewise_fusable op (e.g. conv2d) are in the same group.
+    bool parent_out_ewise = false;
+    bool parent_injective = false;
+    for (const auto& e : inode.inputs) {
+      TOpPattern pt = pattern_vec[e.node_id];
+      if (pt == kOutEWiseFusable) parent_out_ewise = true;
```
Comment (tqchen) [inline nnvm/src/compiler/graph_fuse.cc]: style, multiline if always enclose with {}

**Comment by tqchen** [comment]
Thanks @masahi , this is now merged

**Comment by kazum** [comment]
The below script fails after this commit.  This kind of graph pattern can be found, for example, in NASNet.
```Python
import nnvm
from nnvm import symbol as sym

ch = 3
size = 8
data = sym.Variable(name="data")
concat = sym.concatenate(data, data)
conv = sym.conv2d(data=concat, kernel_size=(1,1), channels=ch*2)
net = sym.elemwise_add(concat, conv)
nnvm.compiler.build(net, "llvm", {"data": (1, ch, size, size)})
```

The result is as follows.
```
Traceback (most recent call last):
  File "tmp.py", line 10, in <module>
    nnvm.compiler.build(net, "llvm", {"data": (1, ch, size, size)})
  File "/home/kazutaka/git/tvm/nnvm/python/nnvm/compiler/build_module.py", line 304, in build
    graph = graph.apply("GraphCompile")
  File "/home/kazutaka/git/tvm/nnvm/python/nnvm/graph.py", line 234, in apply
    check_call(_LIB.NNGraphApplyPasses(self.handle, npass, cpass, ctypes.byref(ghandle)))
  File "/home/kazutaka/git/tvm/nnvm/python/nnvm/_base.py", line 75, in check_call
    raise NNVMError(py_str(_LIB.NNGetLastError()))
nnvm._base.NNVMError: [05:30:54] /home/kazutaka/git/tvm/nnvm/include/nnvm/././op.h:530: Check failed: op != nullptr
```
@masahi, can you check this out?

**Comment by masahi** [comment]
@kazum Thanks! I sent a PR to fix this #1622.

**Pull request merged**